R/test-directory-c.R

```diff
@@ -0,0 +1,356 @@
+
+# +-----------------------------+     +-------------------------------+
+# | Main R process              |     | Subprocess 1                  |
+# | +------------------------+  |     | +---------------------------+ |
+# | | test_dir_parallel()    |  |     | | test_file()               | |
+# | | +-------------------+  |  |     | | +-----------------------+ | |
+# | | | Event loop        |< ------+  | | | SubprocessReporter    | | |
+# | | +-------------------+  |  |  |  | | | +-------------------+ | | |
+# | |    |                   |  |  |  | | | | test_that()       | | | |
+# | |    v                   |  |  |  | | | +-------------------+ | | |
+# | | +-------------------+  |  |  |  | | |    |                  | | |
+# | | | Progress2Reporter |  |  |  |  | | |    v                  | | |
+# | | +-------------------+  |  |  |  | | | +-------------------+ | | |
+# | +------------------------+  |  |--------| signalCondition() | | | |
+# +-----------------------------+  |  | | | +-------------------+ | | |
+#                                  |  | | +-----------------------+ | |
+#                                  |  | +---------------------------+ |
+#                                  |  +-------------------------------+
+#                                  |  +-------------------------------+
+#                                  |--| Subprocess 2                  |
+#                                  |  +-------------------------------+
+#                                  |  +-------------------------------+
+#                                  +--| Subprocess 3                  |
+#                                     +-------------------------------+
+#                                       ...
+#
+# ## Notes
+#
+# * Subprocesses run `callr::r_session` R sessions. They are re-used,
```

Comment:
Maybe integrate these notes into the vignette so they don't even get out of sync? (I think mostly they're there already)

Reply:
I think ideally we would have two vignettes, one for the users, and one with the internals. The current parallel.Rmd is the former, and these notes are really about the implementation details. I can certainly create a `parallel-internals.Rmd` or actually maybe just an `internals.Rmd` with all the testthat internals?